### PR TITLE
BTFSBTT-201: Modify btfs add command to add BTT chunk metadata

### PR DESCRIPTION
**Pull request by steveyeom**
1. "btfs add" command changes
   core/commands/add.go - options
2. Adding metadata functionality:
  core/coreapi/unixfs.go - set up passing metadata json string
  core/coreunix/add.go - convert json string to byte array and verify string validity
3. Dependency resolving code changes to import TRON-US repos.
  All the files but the above files including go.mod

### CODE REVIEW

**Comment by steveyeom** [comment]
Please review this pull request. 
Thank you, 
Steve.

File: core/coreunix/add.go [hunk 1]
```diff
@@ -53,6 +54,7 @@ func NewAdder(ctx context.Context, p pin.Pinner, bs bstore.GCLocker, ds ipld.DAG
 		Pin:        true,
 		Trickle:    false,
 		Chunker:    "",
+		TokenMetadata: "",
```
Comment (taiyangc) [inline core/coreunix/add.go]: go fmt

File: core/coreunix/add.go [hunk 2]
```diff
@@ -425,6 +437,24 @@ func (adder *Adder) addDir(path string, dir files.Directory, toplevel bool) erro
...
+// Convert token metadata in JSON string to byte array in JSON encoding.
+func (adder *Adder) convertMetadataToBytes(checkString bool) ([]byte, error) {
+	s := adder.TokenMetadata
+	b := []byte(s)
+	// Optionally check if the input data is in JSON string.
+	// If not, return zero and error.
+	if checkString {
+		var a interface{}
+		var err error
+		err = json.Unmarshal(b[:len(b)], &a)
```
Comment (taiyangc) [inline core/coreunix/add.go]: Why `b[:len(b)]` instead of just `b`?

File: go.mod [hunk 3]
```diff
@@ -27,7 +30,7 @@ require (
 	github.com/ipfs/dir-index-html v1.0.3
 	github.com/ipfs/go-bitswap v0.1.6
 	github.com/ipfs/go-block-format v0.0.2
-	github.com/ipfs/go-blockservice v0.1.0
+	github.com/ipfs/go-blockservice v0.1.1
```
Comment (taiyangc) [inline go.mod]: Did not revert this back to `v0.1.0`

File: go.mod [hunk 4]
```diff
@@ -6,8 +6,11 @@ require (
 	github.com/FactomProject/btcutilecc v0.0.0-20130527213604-d3a63a5752ec // indirect
 	github.com/Kubuxu/gocovmerge v0.0.0-20161216165753-7ecaa51963cd
 	github.com/StackExchange/wmi v0.0.0-20181212234831-e0a55b97c705 // indirect
-	github.com/TRON-US/go-btfs-cmds v0.1.3
+	github.com/TRON-US/go-btfs-cmds v0.1.4
 	github.com/TRON-US/go-btfs-config v0.1.6
+	github.com/TRON-US/go-mfs v0.2.0
```
Comment (taiyangc) [inline go.mod]: can you run `go mod tidy`? I think `github.com/ipfs/go-mfs` should have been removed? Or is there another package using ipfs/go-mfs as indirect?

File: go.mod [hunk 5]
```diff
@@ -6,8 +6,11 @@ require (
...
+	github.com/TRON-US/go-unixfs v0.3.0
+	github.com/TRON-US/interface-go-btfs-core v0.3.1
```
Comment (taiyangc) [inline go.mod]: Same question about `github.com/ipfs/interface-go-ipfs-core` still not removed

File: core/coreunix/add.go (hunk 1 above)
Comment (steveyeom) [reply core/coreunix/add.go]: I ran "go fmt add.go"

File: go.mod (hunk 4 above)
Comment (steveyeom) [reply go.mod]: I ran. github.com/TRON-US/go-btfs/core is using go-mfs according to "go mod why".

Comment (taiyangc) [reply go.mod]: It should not use `ipfs/go-mfs` anymore. Can you find out which file is still using it?

File: go.mod (hunk 5 above)
Comment (steveyeom) [reply go.mod]: I ran "go mod why" for github.com/ipfs/interface-go-ipfs-core nothing direct. But "go mod tidy" does not remove.

File: go.mod (hunk 4 above)
Comment (steveyeom) [reply go.mod]: No one from go-btfs. 
In go.mod, we have.
github.com/ipfs/go-mfs v0.1.0 // indirect

**Comment by steveyeom** [comment]
Thank you Eric.

**Pull request merged**